File: src/main/java/space/iseki/cmdpipe/EnvVar.java
package space.iseki.cmdpipe;

import kotlin.Pair;
import org.checkerframework.checker.nullness.qual.NonNull;
import org.checkerframework.checker.nullness.qual.Nullable;

public record EnvVar(
        @NonNull String name,
        @Nullable String value,
        boolean confidential
) {
    public EnvVar(Pair<String, String> pair) {
        this(pair.getFirst(), pair.getSecond(), false);
    }

    public EnvVar(@NonNull String name, @Nullable String value) {
        this(name, value, false);
    }

    @Override
    public String toString() {
        return EnvVarFormatter.INSTANCE.format(this);
    }
}

File: src/main/kotlin/space/iseki/cmdpipe/EnvVar.kt
package space.iseki.cmdpipe

data class EnvVar(
    val name: String,
    /**
     * if is null, the variable which matches the name will be removed
     */
    val value: String?,
    val confidential: Boolean = false,
) {
    override fun toString(): String {
        val key = simpleEscapeString(name, true)
        val value = value?.let { simpleEscapeString(it, false) }
        return when {
            value == null -> "$key (cleared)"
            confidential -> "$key ***"
            else -> "$key=\"$value\""
        }
    }

    constructor(pair: Pair<String, String?>) : this(pair.first, pair.second)
}

